zerver/webhooks/newrelic/view.py

```diff
@@ -1,169 +1,264 @@
 from zerver.lib.validator import (
     WildValue,
+    check_float,
     check_int,
     check_list,
     check_none_or,
     check_string,
-    check_string_fixed_length,
+    check_string_in,
     check_union,
 )
 from zerver.lib.webhooks.common import check_send_webhook_message, unix_milliseconds_to_timestamp
 from zerver.models import UserProfile
 
-# Newrelic planned to upgrade Alert Notification Channels to Workflows and Destinations
-# https://discuss.newrelic.com/t/plan-to-upgrade-alert-notification-channels-to-workflows-and-destinations/188205
-# This view will handle both old and new format but will keep it easy to delete the old code
-# once it is EOLed by the end of June, 2023
+PRIORITIES = {
+    "CRITICAL": ":red_circle:",
+    "HIGH": ":orange_circle:",
+    "MEDIUM": ":yellow:",
+    "LOW": ":blue_circle:",
+}
+
+TIME_STAMPS_MAP: dict[str, str] = {
+    "CREATED": "createdAt",
+    "ACTIVATED": "activatedAt",
+    "UPDATED": "updatedAt",
+    "CLOSED": "closedAt",
+    "ACKNOWLEDGED": "acknowledgedAt",
+}
+
+INCIDENT_STATES = ["CREATED", "ACTIVATED", "CLOSED"]
+
+DEFAULT_NEWRELIC_URL = "https://one.newrelic.com/alerts-ai"
+
+INTEGRATION_LAST_UPDATED = "June 2024"
 
-# Once old is EOLed, delete the OPEN_TEMPLATE
-OPEN_TEMPLATE = """
-[Incident]({incident_url}) **opened** for condition: **{condition_name}** at <time:{iso_timestamp}>
-``` quote
-{details}
+BASE_NOTIFICATION_WRAP_TEMPLATE = """
+```spoiler {header}
+{body}
 ```
-""".strip()
+"""
+
+NOTIFICATION_HEADER = "{priority_symbol} {priority} **priority [issue]({incident_url}) has been **{status}** at** <time: {time_updated_str} >"
+
+NOTIFICATION_BODY_TEMPLATE = """
+**[{title}]({incident_url})**
+
+| :file:  **Incident details**  |
+|:--------|
+|:checkbox: Alert policy: {alert_policy_str}|
+|:spiral_notepad: Conditions: {conditions_str}|
+|:warning: Total incidents: **{total_incidents}**|
+|:clock: Incident created at: <time: {time_created_str} > |
+"""
+
+TOPIC_TEMPLATE = "{state} {priority} priority"
+
+FALL_BACK_NOTIFICATION_TEMPLATE = """
+```spoiler :danger: An [incident](https://one.newrelic.com/alerts-ai) updated
+
+**Warning: Unable to send the default notification format**
+At least one essential field is missing from the payload. Try
+to reset the payload template to the default from New Relic
+if you're unsure.
 
-ACTIVE_TEMPLATE = """
-[Incident]({incident_url}) **active** for condition: **{condition_name}** at <time:{iso_timestamp}>
-``` quote
-{details}
+Please reach out to [the Zulip development community](https://chat.zulip.org/#narrow/stream/127-integrations)
+if you think this integration is out-of-date.
+> *Integration last updated: **{integration_last_updated}***
 ```
-""".strip()
+"""
 
-DEFAULT_TEMPLATE = (
-    """[Incident]({incident_url}) **{status}** {owner}for condition: **{condition_name}**""".strip()
-)
+FALL_BACK_TOPIC_TEMPLATE = "Incident alerts"
+
+ZULIP_CUSTOM_FIELDS_KEY = "zulipCustomFields"
```

Comment:
This is another constant that's only used once, so let's just use the string where we expect it.